appengine/pond/js/avatar.js

```diff
@@ -34,22 +34,46 @@ goog.require('BlocklyInterface');
  * Class for a avatar.
  * @param {string} name Avatar's name.
  * @param {string|!Function} code Avatar's code, or generator.
- * @param {!Blockly.utils.Coordinate} startLoc Start location.
  * @param {?number} startDamage Initial damage to avatar (0-100, default 0).
- * @param {!Pond.Battle} battle The battle featuring the avatar.
+ * @param {?Blockly.utils.Coordinate} opt_startLoc Start location.
  * @constructor
  */
-Pond.Avatar = function(name, code, startLoc, startDamage, battle) {
+Pond.Avatar = function(name, code, startDamage, opt_startLoc) {
   this.name = name;
   this.code_ = code;
-  this.startLoc_ = startLoc;
+  this.startLoc_ = opt_startLoc;
   this.startDamage_ = startDamage || 0;
-  this.battle_ = battle;
+  this.battle_ = null;
   this.loc = new Blockly.utils.Coordinate();
-  this.reset();
   console.log(this + ' loaded.');
 };
 
+/**
+ * Given a name create the default player.
+ * @param {string} name The name of the default player.
+ */
+Pond.Avatar.createDefaultAvatar = function(name) {
+  if (name == 'Pond_rookName') {
+    var div = document.getElementById('playerRook');
+    var code = div.textContent;
+  } else if (name == 'Pond_counterName') {
+    var div = document.getElementById('playerCounter');
+    var code = div.textContent;
+  } else if (name == 'Pond_sniperName') {
+    var div = document.getElementById('playerSniper');
+    var code = div.textContent;
+  }
+  return new Pond.Avatar(name, code, 0);
+};
+
+/**
+ * Create an avatar based on the current user's duck.
+ * @param {string} name The name of the avatar.
+ */
+Pond.Avatar.createCurrentAvatar = function(name) {
```

Comment:
I think this should be called create playerAvatar since throughout the game we refer to the current user's duck as the player (and I think its easier to understand).